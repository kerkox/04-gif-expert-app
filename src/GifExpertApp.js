import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

  const initialState = ['One Punch'];
  const [categories, setCategories] = useState(initialState)


  return (
    <>
    <h1>GifExpertApp</h1>
      <AddCategory setCategories={setCategories}/>
    <hr />
    
    <ol>
      {
        categories.map((category,i) => (
          <GifGrid 
            key={category}
            category={category}
          />
        ))
      }
    </ol>
    </>
  )
}


export default GifExpertApp;