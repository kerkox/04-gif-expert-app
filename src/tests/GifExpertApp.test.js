import { shallow } from "enzyme"
import {GifExpertApp} from '../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {
    test('Debe de hacer match con el snapshot', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar una lista de catgegorias', () => {
        const categories = ['Mr Robot', 'Silicon Valley'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
})
