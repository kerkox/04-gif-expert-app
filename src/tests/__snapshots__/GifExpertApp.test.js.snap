// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp/> Debe de mostrar una lista de catgegorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Mr Robot"
      key="Mr Robot"
    />
    <GifGrid
      category="Silicon Valley"
      key="Silicon Valley"
    />
  </ol>
</Fragment>
`;
